app/models/network/conserving_producer.rb

```diff
@@ -5,18 +5,27 @@ class ConservingProducer < Technology
     extend Disableable
 
     def self.disabled?(options)
-      ! options[:flexibility]
+      options[:capping_solar_pv] == 'false'
     end
 
     def self.disabled_class
       Technology
     end
 
-    def conditional_consumption_at(frame)
-      production_at(frame)
+    def production_at(frame)
+      production = super
+      capping = (capacity * capping_percentage).abs
+
+      production > capping ? capping : production
     end
 
     def store(_frame, _amount)
     end
+
+    private
```

Comment:
Last comment for tonight (I swear!).

The normal QI Ruby style is [not to indent](https://github.com/bbatsov/ruby-style-guide#consistent-classes) after calling `private`, `protected`, or `module_function`; just maintain the same level of indentation as before. There's a RuboCop [style config here](https://gist.github.com/antw/c5297c5e7d9138dc83da) – which is mostly consistent with the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) – but you probably don't want to run that now as there'll be too many messages (mostly fixed on that "markets" branch).

I think RuboCop will *always* complain about "inconsistent indentation" if you indent after `private`.